hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/file/AvroFileFormat.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.pipeline.file;
+
+import com.hazelcast.function.BiFunctionEx;
+import com.hazelcast.function.FunctionEx;
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.avro.mapreduce.AvroKeyInputFormat;
+import org.apache.avro.reflect.ReflectData;
+import org.apache.avro.reflect.ReflectDatumReader;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.hadoop.io.NullWritable;
+
+import java.nio.file.Path;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static com.hazelcast.jet.impl.util.Util.uncheckRun;
+
+/**
+ * FileFormat for avro files
+ *
+ *
+ * @param <T>
+ */
+public class AvroFileFormat<T> extends AbstractFileFormat<AvroKey<T>, NullWritable, T>
```

Comment:
we shouldn't use any class(`NullWritable`) from hadoop, it adds a dependency to hadoop even if user wants to read files without hadoop module.